src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`3.cs

```diff
@@ -136,32 +136,20 @@ protected override void SetupNewTextView(IVsTextView textView)
                 {
                     // We need to get our outlining tag source to notify it to start blocking
                     var outliningTaggerProvider = this.Package.ComponentModel.GetService<OutliningTaggerProvider>();
-                    ProducerPopulatedTagSource<IOutliningRegionTag> tagSource;
-                    if (outliningTaggerProvider.TryRetrieveTagSource(wpfTextView, wpfTextView.TextBuffer, out tagSource))
+
+                    // If this file is a metadata-from-source file, we want to force-collapse
+                    if (isOpenMetadataAsSource)
+                    {
+                        outliningManager.CollapseAll(new SnapshotSpan(wpfTextView.TextBuffer.CurrentSnapshot,
+                                                                      start: 0,
+                                                                      length: wpfTextView.TextBuffer.CurrentSnapshot.Length),
+                                                     c => c.Tag.IsImplementation);
+                    }
+                    else
                     {
-                        tagSource.ComputeTagsSynchronouslyIfNoAsynchronousComputationHasCompleted = true;
-
-                        try
-                        {
-                            // If this file is a metadata-from-source file, we want to force-collapse
-                            if (isOpenMetadataAsSource)
-                            {
-                                outliningManager.CollapseAll(new SnapshotSpan(wpfTextView.TextBuffer.CurrentSnapshot,
-                                                                              start: 0,
-                                                                              length: wpfTextView.TextBuffer.CurrentSnapshot.Length),
-                                                             c => c.Tag.IsImplementation);
-                            }
-                            else
-                            {
-                                // Set the initial outlining state by reading from the suo file, this operation requires
-                                // us to synchronously compute the outlining region tags.
-                                viewEx.PersistOutliningState();
-                            }
-                        }
-                        finally
-                        {
-                            tagSource.ComputeTagsSynchronouslyIfNoAsynchronousComputationHasCompleted = false;
```

Comment:
This was basically copied from Paul's change.  I've tested it and it does work properly.  That said, the editor calls into us for accurate tagging.  So if we're slow, it will afffect the editor experience.